File: stu_spring/webflux/webflux/src/main/java/com/first/webflux/error/ErrorCode.java
package com.first.webflux.error;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
public enum ErrorCode {
    TEST_NOT_FOUND("Test Not Found", 404),
    TEST_ALREADY_EXIST("Test Already Exist", 409),
    UNKNOWN_ERROR("Unknown Error", 500);
    private final String error;
    private final int status;
}

File: stu_spring/kotlin_basic/src/main/kotlin/com/kotlin/test/kotlintest/global/exception/enums/ErrorCode.kt
package com.kotlin.test.kotlintest.global.exception.enums

enum class ErrorCode(
    val status: Int,
    val cause: String
) {
    USER_NOT_FOUND(404, "User Not Found"),
    PASSWORD_NOT_MATCH(404, "Password Not Match"),
    USER_ALREADY_EXIST(409, "User Already Exist"),
    INVALID_TOKEN(401, "Invalid Token"),
    TOKEN_EXPIRED(401, "Token Expired"),
    POST_NOT_ACCESSIBLE(403, "Not Accessible"),
    POST_NOT_FOUND(404, "Post Not Found"),
    UNEXPECTED_EXCEPTION(500, "Unexpected Exception")
}